### PR TITLE
BER estimator, K-best detection and channel correlation

### PR DESCRIPTION
**Pull request by BastienTr**
This is the updated PR of #30.

This PR contains:
* An implementation of the K-best Schnorr-Euchner algorithm for MIMO detection.
* A tool to estimate the BER performance of a link model using a Monte Carlo estimator. This relies on a new class named `link_model` that gather all the link informations (modem, channel and receptor).
* New tests on the channel class.
* A new `Modem` method to plot the constellation.
* Helper methods to ease the use of the channel class. These methods allow to set the fading to the most standard fadings (Rayleigh, rician, uncorrelated or exponential correlated).
* Some corrections for the doc and other typos.

### CODE REVIEW

**Comment by BastienTr** [comment]
This PR can now be safely pulled.

File: commpy/links.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            else:
+                received_msg = link_model.receive(channel_output, link_model.channel.channel_gains, link_model.constellation)
+            # Count errors
+            bit_err += (msg != received_msg).sum()  # Remove MIMO padding
+            bit_send += send_chunk
+        BERs[id_SNR] = bit_err / bit_send
+    return BERs
+
+
+class linkModel:
```
Comment (veeresht) [inline commpy/links.py]: @BastienTr @akou97 Most of the PR looks good. Thank you! Small request on the naming conventions. Can we stick to using PascalCase / UpperCamelCase for class names e.g., linkModel would become "LinkModel" 
and snake_case for function (including member functions) names and variables.

File: commpy/modulation.py [hunk 2]
```diff
@@ -84,22 +88,64 @@ def demodulate(self, input_symbols, demod_type, noise_var = 0):
...
+                            llr_den = llr_den + exp(
+                                (-abs(current_symbol - self.constellation[const_index]) ** 2) / noise_var)
+                    demod_bits[i * self.num_bits_symbol + self.num_bits_symbol - 1 - bit_index] = log(llr_num / llr_den)
         else:
             pass
             # throw an error
 
         return demod_bits
 
+    def plotCons(self):
```
Comment (veeresht) [inline commpy/modulation.py]: @BastienTr @akou97 Same comment about naming conventions here. for e.g. plotCons would become plot_constellation.

File: commpy/examples/plotConsModem.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+from commpy.modulation import PSKModem, QAMModem
+
+# =============================================================================
+# Example constellation plot of Modem
+# =============================================================================
+
+# Constellation corresponding to PSKModem for 4 bits per symbols
+psk = PSKModem(16)
+psk.plotCons()
```
Comment (veeresht) [inline commpy/examples/plotConsModem.py]: @BastienTr Seems like you forgot to update this to ```plot_constellation()```

File: commpy/examples/plotConsModem.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+# Constellation corresponding to QAMModem for 2 bits per symbols
+qam = QAMModem(4)
+qam.plotCons()
```
Comment (veeresht) [inline commpy/examples/plotConsModem.py]: @BastienTr Seems like you forgot to update this to ```plot_constellation()```

**Comment by BastienTr** [comment]
My bad. Thanks for the look @veeresht :+1:

**Pull request merged**